scrapers/csci_topics_scraper/main.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# Python standard library
+import asyncio
+from operator import itemgetter
+import os
+import re
+import json
+import sys
+
+# External dependnecies
+import aiohttp
+import bs4
+import requests
+
+BeautifulSoup = lambda data: bs4.BeautifulSoup(data, features="lxml")
+
+
+async def get_topics_txts():
+    global session
+    url = "https://www.cs.rpi.edu/~goldsd/docs/"
+    async with session.get(url) as request:
+        soup = BeautifulSoup(await request.text())
+        links = soup.find("table").findAll("a")
+        out = []
+        for link in links:
+            if "topics-courses" in link["href"]:
+                out.append(link["href"])
+        # sorts reverse chronologically
+        return sorted(out, reverse=True, key=lambda x: (x[1:3] + str.lower(x[0])))
+
+
+def goldschmidt_term_to_sis_term(term):
+    term = term.lower()
+    if term[0] == "s":
+        postfix = "01"
+    elif term[0] == "u":
+        postfix = "05"
+    elif term[0] == "f":
+        postfix = "09"
+    else:
+        # This shouldn't be possible
+        raise Exception(f"Invalid term code {term}")
+    return "20" + term[1:] + postfix
+
+
+async def scrape_txt(filename, list_of_terms):
+    global session
+    url = f"https://www.cs.rpi.edu/~goldsd/docs/{filename}"
+    async with session.get(url) as request:
+        data = re.split(r"\n======*", (await request.text()))
+        for term_data in data:
+            # Goldschmidt uses Windows which is why we have to deal with \r,
+            # which I do by just removing all of them. Then we split on
+            # double-newlines, because each course "block" is split by these.
+            # Sometimes course "blocks" are split by more than 2 newlines
+            # so this results in empty or whitespace-only strings, which is
+            # why "if line.strip()" is there, to get rid of them.
+            courses = [
+                line
```

Comment:
Should this be `line.strip()`?

Reply:
Probably. I'll add that, though it worked without it.